File: ruby/sieve/sieve.rb
module BookKeeping
	VERSION = 1
end

class Sieve
	def initialize n
		@limit = n
		@prime_array = []
	end

	def primes
		(1..@limit).each do |num|
			is_prime? num
		end
		@prime_array
	end

	def is_prime? num
		return false if num == 1
		unless @prime_array.empty?
			@prime_array.each do |prime|
				return false if num % prime == 0
			end
		end
		@prime_array.push num
	end
end

File: python/sieve/sieve.py
def sieve(input):
	primes = [None, None] + [True] * (input - 1)
	for p in range(2, input+1):
		if primes[p]:
			for m in range(2*p,input+1,p):
				primes[m] = None
	return [i for i, p in enumerate(primes) if p]

# def sieve(input):
# 	primes = range(2,input+1)
# 	for p in primes:
# 		primes = filter(lambda x: x==p or x % p != 0, primes)
# 	return primes